pytorch_lightning/core/hooks.py

```diff
@@ -310,6 +310,13 @@ def on_post_move_to_device(self):
 
         """
 
+    def on_model_parallel_setup(self) -> None:
+        """
+        Hook to create modules in a distributed aware context. This is useful for when using sharded plugins,
+        where we'd like to shard the model instantly, which is useful for extremely large models
+        which can save memory and initialization time.
+        """
```

Comment:
this should specify whether it runs on fit/val/test/predict stages. otherwise it can be confusing why this is different from the existing `setup` hook

Reply:
Good idea, I'll update the docstring as to why this doesn't implement fit/val/test/predict stages